src/Microsoft.SqlTools.ServiceLayer/LanguageServices/Completion/CompletionService.cs

```diff
@@ -0,0 +1,166 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Collections.Generic;
```

Comment:
just a general comment is that there's quite a bit of refactoring and probably some functional changes as well, this code is pretty hard to test all the various cases (timeouts, errors, etc.).  if we want to take for preview #2 release let's get this in and spend some focused testing time to make sure nothing regressed.